jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerIntegrationClientsTest.java

```diff
@@ -93,83 +103,106 @@ public void testServerTemplateClients() {
 
-        assertEquals(1,
-                     kieServerIntegration.getServerTemplatesClients().size());
-        assertServerInstanceFailedEndpoint(kieServerIntegration,
-                                           serverInstance2);
-        assertContainerFailedEndpoint(kieServerIntegration,
-                                      serverInstance2,
-                                      containerSpec.getId());
+        assertEquals(1, kieServerIntegration.getServerTemplatesClients().size());
+        assertServerInstanceFailedEndpoint(kieServerIntegration, serverInstance2);
+        assertContainerFailedEndpoint(kieServerIntegration, serverInstance2, containerSpec.getId());
 
-        assertEquals(0,
-                     kieServerIntegration.getServerInstancesById().size());
+        assertEquals(0, kieServerIntegration.getServerInstancesById().size());
         assertNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId1));
         assertNull(kieServerIntegration.getServerInstancesById().get(serverInstanceId2));
 
         kieServerIntegration.onServerTemplateDeleted(new ServerTemplateDeleted(serverTemplateId));
-        assertEquals(0,
-                     kieServerIntegration.getServerTemplatesClients().size());
+        assertEquals(0, kieServerIntegration.getServerTemplatesClients().size());
+    }
+
+    @Test
+    public void testCreateAvailableClients() throws Exception {
+        final String serverTemplateId = "kie-server-test";
+        final String serverInstanceId1 = serverTemplateId + "@1";
+        final String serverInstanceId2 = serverTemplateId + "@2";
+        final ContainerSpec containerSpec = newContainerSpec();
+        final ServerInstance serverInstance1 = new ServerInstance();
+        serverInstance1.setServerTemplateId(serverTemplateId);
+        serverInstance1.setUrl("http://1");
+
+        serverInstance1.setServerInstanceId(serverInstanceId1);
+        final ServerTemplate serverTemplate = new ServerTemplate(serverTemplateId, serverTemplateId);
+        serverTemplate.addContainerSpec(containerSpec);
+
+        final ServerInstance serverInstance2 = new ServerInstance();
+        serverInstance2.setServerTemplateId(serverTemplateId);
+        serverInstance2.setUrl("http://2");
+        serverInstance2.setServerInstanceId(serverInstanceId2);
+
+        serverTemplate.addServerInstance(serverInstance2);
+
+        when(specManagementService.listServerTemplates()).thenReturn(new ServerTemplateList(Collections.singletonList(serverTemplate)));
+
+        CountDownLatch latch = new CountDownLatch(2);
+
+        doAnswer(a -> {
+            final Object result = a.callRealMethod();
+            latch.countDown();
+            return result;
+        }).when(kieServerIntegration).createClientForTemplate(any(), any(), any());
+
+        kieServerIntegration.createAvailableClients();
+
+        assertTrue(latch.await(10, TimeUnit.SECONDS));
```

Comment:
please add assertion messages (at least here, describing why the latch is here.)

Reply:
done